libs/components/datetime/src/lib/modules/datepicker/datepicker.component.ts

```diff
@@ -429,4 +447,24 @@ export class SkyDatepickerComponent implements OnDestroy, OnInit {
       this.#customDatesSubscription = undefined;
     }
   }
+
+  #populateInputBoxHelpText(): void {
+    /* Ternary is a safety check */
+    const hintTextObs: Observable<string | undefined> = this.dateFormat
+      ? this.#resourceSvc.getString(
+          'skyux_datepicker_format_hint_text',
+          this.dateFormat,
+        )
+      : /* istanbul ignore next */ of('');
+
+    this.#hintTextResourceUnsubscribe.next();
+
+    hintTextObs
+      .pipe(takeUntil(this.#hintTextResourceUnsubscribe))
```

Comment:
This logic is fairly complex. Curious, did you try using `sky-format` for this? https://developer.blackbaud.com/skyux/components/format?docs-active-tab=code-examples

Reply:
I'll be honest - I'm not sure what logic you are calling out as being complex here? The line that this comment is on is just the `pipe` line - and then the contents is just a simple if statement and a function call to the host service.